### PR TITLE
Add validation for KUBE_USER

### PR DESCRIPTION
**Pull request by adityakali**
Malformed KUBE_USER causes error in cluster setup.

cc/ @kubernetes/goog-image 
@Q-Lee @Amey-D  Can you please review?

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31318)

### CODE REVIEW

**Comment by adityakali** [comment]
This is needed in next 1.3.x cut as well. Can someone please set appropriate labels for it?
Thanks.

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -752,10 +752,10 @@ function start-kube-apiserver {
   fi
   local -r src_dir="${KUBE_HOME}/kube-manifests/kubernetes/gci-trusty"
 
-  if [[ -n "${KUBE_USER:-}" ]]; then
+  if [[ -n "${KUBE_USER}" ]]; then
```
Comment (rmmh) [inline cluster/gce/gci/configure-helper.sh]: This should still be `"${KUBE_USER:-}"` if this script is supposed to work without KUBE_USER being set.

Comment (adityakali) [inline cluster/gce/gci/configure-helper.sh]: KUBE_USER is now initialized early on in the setup (on line 1119) if its not set. So its redundant to do it here.

**Comment by adityakali** [comment]
Updated the PR with additional checks. PTAL.

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (rmmh) [inline cluster/gce/gci/configure-helper.sh]: No, `"${KUBE_USER:-}"` doesn't set KUBE_USER. Use `${KUBE_USER:=admin}` if you want to initialize it.

Comment (Q-Lee) [inline cluster/gce/gci/configure-helper.sh]: Please fix this under trusty as well.

Comment (adityakali) [inline cluster/gce/gci/configure-helper.sh]: > Please fix this under trusty as well.

The change in cluster/common.sh takes care of trusty too, no?

Comment (adityakali) [inline cluster/gce/gci/configure-helper.sh]: > No, ${KUBE_USER:-} doesn't set KUBE_USER. Use ${KUBE_USER:=admin} if you want to initialize it.

Ah. I assumed it also sets it. I restored it as before. But I think with `set -o nounset` at the top of this file, even `${KUBE_USER:-}` will cause this code to fail if KUBE_USER is unset. I also see KUBE_USER used with the not-null check above.

Comment (spxtr) [inline cluster/gce/gci/configure-helper.sh]: With nounset, `${THIS_IS_UNSET:-}` will not fail. [This is a super-useful resource](http://tldp.org/LDP/abs/html/parameter-substitution.html).

**Pull request merged**